src/jit/gentree.h

```diff
@@ -4037,26 +3884,198 @@ struct GenTreeIndir : public GenTreeOp
 #endif
 };
 
-// gtObj  -- 'object' (GT_OBJ). */
+// gtBlk  -- 'block' (GT_BLK, GT_STORE_BLK).
+//
+// This is the base type for all of the nodes that represent block or struct
+// values.
+// Since it can be a store, it includes gtBlkOpKind to specify the type of
+// code generation that will be used for the block operation.
+
+struct GenTreeBlk : public GenTreeIndir
+{
+public:
+    // The data to be stored (null for GT_BLK)
+    GenTree*& Data()
+    {
+        return gtOp2;
+    }
+    void SetData(GenTree* dataNode)
+    {
+        gtOp2 = dataNode;
+    }
+
+    // The size of the buffer to be copied.
+    unsigned Size() const
+    {
+        return gtBlkSize;
+    }
+
+    unsigned gtBlkSize;
+
+    // Return true iff the object being copied contains one or more GC pointers.
+    bool HasGCPtr();
+
+    // True if this BlkOpNode is a volatile memory operation.
+    bool IsVolatile() const
+    {
+        return (gtFlags & GTF_BLK_VOLATILE) != 0;
+    }
+
+    // True if this BlkOpNode is a volatile memory operation.
+    bool IsUnaligned() const
+    {
+        return (gtFlags & GTF_BLK_UNALIGNED) != 0;
+    }
+
+    // Instruction selection: during codegen time, what code sequence we will be using
+    // to encode this operation.
+    enum
+    {
+        BlkOpKindInvalid,
+        BlkOpKindHelper,
+        BlkOpKindRepInstr,
+        BlkOpKindUnroll,
+    } gtBlkOpKind;
+
+    bool gtBlkOpGcUnsafe;
```

Comment:
As per my understanding a GT_BLK node doesn't contain gc-ptrs and hence whether it is GcUnsafe will always be false.  This makes me wonder, whether this member belongs to GenTreeObj instead of GenTreeBlk?  Essentially, are GenTreeBlk and GenTreeDynBlk nodes unnecessarily paying the space penalty of having gtBlkOpGcUnsafe member?

We add a member method in base class GenTreeBlk::IsGCUnsafe() - that will return false for GT_BLK/DYN_BLK and gtBlkOpGCUnsafe for GT_OBJ.